### PR TITLE
feat(coingecko): add function for coin history endpoint

### LINKED ISSUE #24: Add a function for /coins/{id}/history coingecko endpoint
For more details look at #4

### PR DESCRIPTION
**Pull request by codeproline**
This PR adds functions for coingecko's `/coins/{id}/history` endpoint.

Closes #24

### CODE REVIEW

File: packages/coingecko/src/query/routes/coinHistory.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  Input_coinHistory,
+  CoinHistory,
+  HTTP_UrlParam,
+  HTTP_Query,
+  HTTP_ResponseType,
+} from "../w3";
+
+export function coinHistory(input: Input_coinHistory): CoinHistory {
+  const url = COINGECKO_API_URL + "/coins/" + input.id + "/history";
+  const urlParams: Array<HTTP_UrlParam> = [{ key: "date", value: input.date }];
```
Comment (Niraj-Kamdar) [inline packages/coingecko/src/query/routes/coinHistory.ts:23]: Can you add a validator here to check whether date is in valid format? Ref: https://www.assemblyscript.org/stdlib/date.html#constructor

**Review by Niraj-Kamdar — CHANGES_REQUESTED** [review]
Thanks for adding more test coverage! It would be nice to validate inputs whenever possible before making the network request to save time.

**Comment by Niraj-Kamdar** [comment]
Please merge master into your PR to make it up-to-date with master

File: packages/coingecko/src/query/routes/coinHistory.ts (hunk 1 above)
Comment (codeproline) [reply packages/coingecko/src/query/routes/coinHistory.ts:29]: Thanks for the feedback, I've added validation for date there.

**Review by Niraj-Kamdar — CHANGES_REQUESTED** [review]
Add a separate test for date validation fail. Use different argument and make sure it works on edge cases like leap year, etc.

**Comment by codeproline** [comment]
> Add a separate test for date validation fail. Use different argument and make sure it works on edge cases like leap year, etc.

@Niraj-Kamdar I've added new test case for testing validation fail. Also, used `Date` object to make sure input is valid.

File: packages/coingecko/src/query/routes/coinHistory.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  Input_coinHistory,
+  CoinHistory,
+  HTTP_UrlParam,
+  HTTP_Query,
+  HTTP_ResponseType,
+} from "../w3";
+
+export function coinHistory(input: Input_coinHistory): CoinHistory {
+  if (!isValidDateString(input.date)) {
+    throw new Error("invalid date");
```
Comment (Niraj-Kamdar) [inline packages/coingecko/src/query/routes/coinHistory.ts:25]:
```
    throw new Error("Invalid date format! Use dd-mm-yyyy format.");
```

File: packages/coingecko/src/__tests__/e2e/integration.spec.ts [hunk 3]
```diff
@@ -123,4 +128,88 @@ describe("Coingecko", () => {
...
+    expect(developerData?.code_additions_deletions_4_weeks?.additions).toBe(0);
+
+    // check public interest stats
+    const publicInterestStats = result.data?.coinHistory.public_interest_stats;
+    expect(publicInterestStats?.alexa_rank).toBe(33166);
+    expect(publicInterestStats?.bing_matches).toBeNull();
+  });
+
+  it("should throw error when date is not valid", async () => {
+    const id = "tron";
```
Comment (Niraj-Kamdar) [inline packages/coingecko/src/__tests__/e2e/integration.spec.ts:197]: Add tests for various failing cases like: 
`32-10-2020` - invalid date (upper boundary)
`00-10-2020` - invalid date (lower boundary)
`29-02-2018` - not leap year(special cases)
`01-13-2020` - invalid month
`01/02/2020` -invalid format

Since validation is critical part and if it doesn't work user may have to make unnecessary network request or in worst case won't be able to query some valid date.

**Review by Niraj-Kamdar — CHANGES_REQUESTED** [review]

File: packages/coingecko/src/__tests__/e2e/integration.spec.ts (hunk 3 above)
Comment (codeproline) [reply packages/coingecko/src/__tests__/e2e/integration.spec.ts]: Thanks @Niraj-Kamdar, I added them.

**Review by Niraj-Kamdar — APPROVED** [review]
LGTM!
